packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,311 @@
+          'Adding a SensitiveContent widget with ContentSensitivity $sensitivityLevel is unsupported by ContentSensitivitySetting',
+        );
+    }
+  }
+
+  /// Decreases the count of [SensitiveContent] widgets with [sensitivityLevel] set.
+  void removeWidgetWithContentSensitivity(ContentSensitivity sensitivityLevel) {
+    switch (sensitivityLevel) {
+      case ContentSensitivity.sensitive:
+        _sensitiveWidgetCount--;
+      case ContentSensitivity.autoSensitive:
+        _autoSensitiveWidgetCount--;
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount--;
+      // ignore is safe because it protects this setting from tracking SensitiveContent
+      // widgets with an _unknown ContentSensitivity. _unknown is private to avoid
+      // developers using it as a SensitiveContent sensitivity level.
+      // ignore: no_default_cases
+      default:
+        throw FlutterError(
+          'Removing a SensitiveContent widget with ContentSensitivity $sensitivityLevel is unsupported by ContentSensitivitySetting',
+        );
+    }
+  }
+
+  /// Returns true if this class is currently tracking at least one [SensitiveContent] widget.
+  bool get hasWidgets =>
+      max(0, _sensitiveWidgetCount) +
+          max(0, _autoSensitiveWidgetCount) +
+          max(0, _notSensitiveWigetCount) >
+      0;
+
+  /// Returns the highest prioritized [ContentSensitivity] level of the [SensitiveContent] widgets
+  /// that this setting tracks.
+  ContentSensitivity? get contentSensitivityBasedOnWidgetCounts {
+    if (_sensitiveWidgetCount > 0) {
+      return ContentSensitivity.sensitive;
+    }
+    if (_autoSensitiveWidgetCount > 0) {
+      return ContentSensitivity.autoSensitive;
+    }
+    if (_notSensitiveWigetCount > 0) {
+      return ContentSensitivity.notSensitive;
+    }
+    return null;
+  }
+}
+
+/// Host of the current content sensitivity level for the widget tree that contains
+/// some number [SensitiveContent] widgets.
+@visibleForTesting
+class SensitiveContentHost {
+  SensitiveContentHost._();
+
+  bool? _contentSenstivityIsSupported;
+  late final ContentSensitivitySetting _contentSensitivitySetting = ContentSensitivitySetting();
+  ContentSensitivity? _fallbackContentSensitivitySetting;
+
+  final SensitiveContentService _sensitiveContentService = SensitiveContentService();
+
+  /// The [ContentSensitivity] level set for the entire widget tree calculated from
+  /// [_contentSensitivitySetting].
+  @visibleForTesting
+  ContentSensitivity? calculatedContentSensitivityLevel;
+
+  /// [SensitiveContentHost] instance for the widget tree.
+  @visibleForTesting
+  static final SensitiveContentHost instance = SensitiveContentHost._();
+
+  /// The state of content sensitivity in the widget tree.
+  ///
+  /// Contains the number of widgets with each [ContentSensitivity] level and
+  /// the current [ContentSensitivity] setting.
+  @visibleForTesting
+  ContentSensitivitySetting? getContentSenstivityState() {
```

Comment:
This could be an actual getter: `ContentSensitivitySetting? get contentSenstivityState {`